src/bosh-softlayer-cpi/softlayer/pool/softlayer_pool_deleter.go

```diff
@@ -50,6 +50,17 @@ func (c *softLayerPoolDeleter) Delete(cid int) error {
 				PublicVlan:  int32(virtualGuest.PrimaryNetworkComponent.NetworkVlan.Id),
 				State:       models.StateFree,
 			}
+
+			service, err := c.softLayerClient.GetSoftLayer_Virtual_Guest_Service()
+			if err != nil {
+				return bosherr.WrapErrorf(err, "Powering off VirtualGuest `%d`", cid)
+			}
+
+			service.PowerOffSoft(cid)
```

Comment:
Why is this statement not: `err = service.PowerOffSoft(cid)`? as it stands you are not catching the error which also tells me you don't have enough tests for this part of the code. Please fix both.